packages-next/keystone/src/session/index.ts

```diff
@@ -78,38 +77,28 @@ export function withItemData(createSession: CreateSession, fieldSelections: Fiel
     const { get, ...sessionStrategy } = createSession();
     return {
       ...sessionStrategy,
-      get: async ({ req, system }) => {
-        const session = await get({ req, system });
+      get: async ({ req, createContext }) => {
+        const session = await get({ req, createContext });
+        const sudoContext = createContext({ skipAccessControl: true });
         if (
           !session ||
           !session.listKey ||
           !session.itemId ||
-          !system.adminMeta.lists[session.listKey]
+          !sudoContext.lists[session.listKey]
         ) {
           return;
         }
-        const context = system.createContext({ skipAccessControl: true });
-        const { gqlNames } = system.adminMeta.lists[session.listKey];
         // If no field selection is specified, just load the id. We still load the item,
         // because doing so validates that it exists in the database
-        const fields = fieldSelections[session.listKey] || 'id';
-        const query = parse(`query($id: ID!) {
-          item: ${gqlNames.itemQueryName}(where: { id: $id }) {
-            ${fields}
-          }
-        }`);
-        const args = { id: session.itemId };
-        const result = await execute(system.graphQLSchema, query, null, context, args);
-        // TODO: This causes "not found" errors to throw, which is not what we want
-        // TODO: Also, why is this coming back as an access denied error instead of "not found" with an invalid session?
```

Comment:
Separate topic: I think we bungled this erroring stuff.

Right now we can't return errors because they're misleading. A "not found" comes back as "you don't have access" which leads to questions "what have I done wrong that I don't have access?".

When I say we bungled it, I mean "you don't have access" should come back as "this does not exist" and not the other way around. In my experience that's how every other system handles it when the difference is not technically discernible. And I say this because I've spent time recently debugging what I'm doing wrong that the code I wrote doesn't have the right access control on it, until I learned the quirk.

(didn't really mean to start this conversation here, just this comment was my marker for it, not the right place for that, but we should backlog it elsewhere)

cc/ @molomby to put this on your radar too